python/pyarrow/_dataset_parquet.pyx

```diff
@@ -56,9 +62,113 @@ from pyarrow._parquet cimport (
 
 cdef Expression _true = Expression._scalar(True)
 
-
 ctypedef CParquetFileWriter* _CParquetFileWriterPtr
 
+IF PARQUET_ENCRYPTION_ENABLED:
+
+    from pyarrow._parquet_encryption cimport *
+
+    cdef class ParquetEncryptionConfig(_Weakrefable):
+        """
+        Configuration for Parquet Encryption.
+
+        Parameters
+        ----------
+        crypto_factory : object
+            Factory for creating cryptographic instances.
+        kms_connection_config : object
+            Configuration for connecting to Key Management Service.
+        encryption_config : object
+            Configuration for encryption settings.
```

Comment:
Could we add specific objects in parameters for `ParquetEncryptionConfig` and `ParquetDecryptionConfig`, like `pyarrow.parquet.encryption.CryptoFactory` or `CryptoFactory` instead of just object?